client/state/themes/actions.js

```diff
@@ -765,3 +768,44 @@ export function confirmDelete( themeId, siteId ) {
 		);
 	};
 }
+
+function setPreviewingThemeData( themeId, siteId ) {
+	return {
+		type: PREVIEWING_THEME_DATA,
+		themeData: {
+			themeId,
+			siteId
+		}
+	};
+}
+
+export function setPreviewOptions( primary, secondary ) {
+	return {
+		type: PREVIEWING_THEME_OPTIONS,
+		themeOptions: {
+			primary,
+			secondary
+		}
+	};
+}
+
+function openThemePreview() {
+	return {
+		type: THEME_PREVIEW_STATE,
+		isEnabled: 'enabled'
+	};
+}
+
+export function closeThemePreview() {
+	return {
+		type: THEME_PREVIEW_STATE,
+		isEnabled: 'disabled'
+	};
+}
+
+export function requestPreview( themeId, siteId ) {
```

Comment:
Nomenclature nit: I'd like if we used `request` only for network actions that `GET` data (so e.g. `requestThemes` and `requestActiveTheme`) but neither for `POST` actions (`activateTheme` instead of `requestThemeActivation`) nor for non-network actions 😄 Just `preview`, maybe? Or `showPreview` (if `showThemePreview` is already taken, as I'm suggesting above)?